vortex-layout/src/layouts/stats/reader.rs

```diff
@@ -0,0 +1,167 @@
+use std::ops::Range;
+use std::sync::{Arc, RwLock};
+
+use async_once_cell::OnceCell;
+use vortex_array::aliases::hash_map::HashMap;
+use vortex_array::stats::{stats_from_bitset_bytes, Stat};
+use vortex_array::ContextRef;
+use vortex_dtype::TryFromBytes;
+use vortex_error::{vortex_bail, vortex_err, vortex_panic, VortexExpect, VortexResult};
+use vortex_expr::pruning::PruningPredicate;
+use vortex_expr::{ExprRef, Identity};
+use vortex_mask::Mask;
+
+use crate::layouts::stats::stats_table::StatsTable;
+use crate::layouts::stats::StatsLayout;
+use crate::reader::LayoutReader;
+use crate::scan::ScanExecutor;
+use crate::{ExprEvaluator, Layout, LayoutVTable, RowMask};
+
+type PruningCache = Arc<OnceCell<Option<Mask>>>;
+
+#[derive(Clone)]
+pub struct StatsReader {
+    layout: Layout,
+    ctx: ContextRef,
+    executor: Arc<ScanExecutor>,
+
+    /// The number of blocks
+    nblocks: usize,
+    /// The size of each block (except possibly the last)
+    block_size: usize,
```

Comment:
rows? bytes?

Reply:
Doh, should be len for #rows, and size for #bytes